planner/type_join.go

```diff
@@ -408,8 +413,7 @@ func (n *typeJoinOne) valuesPrimary(doc core.Doc) core.Doc {
 
 	// re-initialize the sub type plan
 	if err := n.subType.Init(); err != nil {
-		log.ErrorE(n.p.ctx, "Sub-type initialization error at scan node reset", err)
-		return doc
+		return doc, errors.Wrap("sub-type initialization error at scan node reset", err)
```

Comment:
todo: After adding the above to `errors.go`, you can use the defined error function here :)

Reply:
sorted